src/NexusMods.Media/ImageResizer.cs

```diff
@@ -7,38 +7,69 @@ public sealed class ImageResizer<TResourceIdentifier> : ANestedResourceLoader<TR
     where TResourceIdentifier : notnull
 {
     private readonly SKSizeI _newSize;
+    private readonly bool _maintainAspectRatio;
 
     public ImageResizer(
         SKSizeI newSize,
-        IResourceLoader<TResourceIdentifier, SKBitmap> innerLoader) : base(innerLoader)
+        IResourceLoader<TResourceIdentifier, SKBitmap> innerLoader,
+        bool maintainAspectRatio = true) : base(innerLoader)
     {
         _newSize = newSize;
+        _maintainAspectRatio = maintainAspectRatio;
     }
 
     protected override ValueTask<Resource<SKBitmap>> ProcessResourceAsync(
         Resource<SKBitmap> resource,
         TResourceIdentifier resourceIdentifier,
         CancellationToken cancellationToken)
     {
-        return ValueTask.FromResult(resource with
+        var original = resource.Data;
+
+        var targetSize = _newSize;
+
+        /*
+         * If Width is larger, the Height is calculated. If Height is larger, the Width is calculated. This ensures the resizing
+         * is based on the dominant dimension while maintaining the aspect ratio
+        */
+        if (_maintainAspectRatio)
         {
-            Data = resource.Data.Resize(_newSize, quality: SKFilterQuality.Low),
-        });
+            var aspectRatio = (float)original.Height / original.Width;
+
+            if (_newSize.Width > _newSize.Height)
+            {
+                var newHeight = (int)(_newSize.Width * aspectRatio);
+                targetSize = new SKSizeI(_newSize.Width, newHeight);
+            }
+            else
+            {
+                var newWidth = (int)(_newSize.Height / aspectRatio);
+                targetSize = new SKSizeI(newWidth, _newSize.Height);
+            }
```

Comment:
This currently doesn't take into account the case where the original major axis doesn't match the target major axis, potentially resulting in images that have greater dimensions than the targets.

This alternative should handle that as well:
```suggestion
            var originalAspectRatio = (float)original.Width / original.Height;
            var targetAspectRatio = (float)_newSize.Width / _newSize.Height;
    
            if (originalAspectRatio > targetAspectRatio)
            {
                // Original is wider compared to target - width becomes the limiting factor
                targetSize = new SKSizeI(_newSize.Width, (int)(_newSize.Width / originalAspectRatio));
            }
            else
            {
                // Original is taller compared to target - height becomes the limiting factor
                targetSize = new SKSizeI((int)(_newSize.Height * originalAspectRatio), _newSize.Height);
            }
```